spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/config/MongoDbInboundChannelAdapterParserTests-context.xml

```diff
@@ -28,6 +28,33 @@
 		<int:poller fixed-rate="100"/>
 	</int-mongodb:inbound-channel-adapter>
 
+	<int-mongodb:inbound-channel-adapter id="fullConfigWithQueryExpression"
+						collection-name-expression="'foo'"
+						query-expression="new org.springframework.data.mongodb.core.query.BasicQuery('{''address.state'' : ''PA''}').limit(2)"
```

Comment:
Take a look, please, into the `TwitterSearchOutboundGateway.onInit()` and pay attention to the 
```
((StandardTypeLocator) typeLocator).registerImport("org.springframework.social.twitter.api");
```
The same we can do here in the `MongoDbMessageSource` to avoid this long package string in the queries.